### DISCUSSION TITLE
QTable pagination

### DISCUSSION THREAD

**Discussion by LadIQe** [quasarframework/quasar#16595] [General - Components / Directives / etc]
Hello, Im finally working with Vue 3 and Quasar after 1 year (had to work with angular - oh my).

Anyway, Im facing some difficulties using QTable with API sorting, filtering and pagination.

Im using Pinia (setup store) to store my data, pagination and filters (not sure if its good approach, but hey, my head is still somehow in angular ways)
```
const state = reactive<ReceiverStore>({
    data: [],
    pagination: {
      page: 1,
      rowsPerPage: 5,
      rowsNumber: 0
    },
    filters: {
      type: null,
      from: null,
      to: null
    },
    isLoading: false
  })
```

this state is returned like this
```
return {
    ...toRefs(state),
    fetchList,
    setFilters
  }
```

Now my question, store is initialized in page component and its data are passed down through props to child components. One of them is table component, which has QTable, filters etc.

It is possible to use pagination and filters directly from props like this? Because, even my API calls are correct (per page is working), pagination in QTable is not updated (only when first time rendered)
```
<script>
const props = defineProps<{
  data: ReceiverItem[]
  pagination: Pagination
  filters: ReceiverFilters
  isLoading: boolean
}>()

const pagination = toRef(props, 'pagination')
const internalFilters = ref({ ...props.filters })
</script>

<template>
  <q-table
    :pagination="pagination"
    flat
    row-key="id"
    :title="t('receiverTable.name')"
    :columns="columns"
    :rows="props.data"
    :loading="props.isLoading"
    :rows-per-page-options="[5, 10, 25, 50, 100]"
    :filter="internalFilters"
    @request="tableRequest"
  >
    <template #top-right>
      <q-select
        v-model="internalFilters.type"
        clearable
        :label="t('receiverTable.filters.typeSelectPlaceholder')"
        :options="typeOptions"
        style="width: 150px"
      />
    </template>
  </q-table>
</template>
```

Sorry for anti-patterns (if any), like I said, it was long time.

Thank you

edit: need to mention, filters are also "stored" in url, so you can send whole url with query filters to someone else and table should show same result (filters are directly pushed to url, then router pass this query as props to page component, where is watcher on query params, which in results store filters to Pinia store and call fetch)

**Comment by metalsadman** [comment]
in serverside pagination, you need to update the pagination properties in your request handler after it resolves, like what you see here https://quasar.dev/vue-components/table#example--synchronizing-with-server

**Reply by LadIQe** [reply]
yes, my pagination is updated in pinia, I grab all needed data in table request and emit them to the parent component, which is doing its job

```
const tableRequest: QTableProps['onRequest'] = ({ pagination, filter }) => {
  const type = filter.type as ReceiverType | null
  const from = filter.from as string | null
  const to = filter.to as string | null

  emit('filter-change', {
    page: pagination.page,
    rowsPerPage: pagination.rowsPerPage,
    type: type || undefined,
    from: from || undefined,
    to: to || undefined
  })
}
```

so all updating filters, pagination is done in pinia before API request and after it. But pagination in QTable is not updated even I have actual data, you have to use
`v-model:pagination="pagination"`

which I dont like to be honest
